perfkitbenchmarker/providers/mesos/mesos_docker_instance.py

```diff
@@ -0,0 +1,206 @@
+    tasks_running = output['app']['tasksRunning']
+    if not tasks_running:
+      raise Exception("Container is not booted yet. Retrying.")
+
+  @vm_util.Retry(poll_interval=10, max_retries=100, log_errors=True)
+  def _SetupSSH(self):
+    """
+    Setup SSH connection details for each instance:
+    - IP address of the instance is the address of a host which instance
+    is running on,
+    - SSH port is drawn by Marathon and is unique for each instance.
+    """
+    output = requests.get(self.app_url)
+    output = json.loads(output.text)
+    tasks = output['app']['tasks']
+    if not tasks or not tasks[0]['ports']:
+      raise Exception("Unable to figure out where the container is running."
+                      "Retrying to retrieve host and port.")
+    self.ip_address = tasks[0]['host']
+    self.ssh_port = tasks[0]['ports'][0]
+    internal_ip, _ = self.RemoteCommand("ifconfig eth0 | grep 'inet addr' | awk"
+                                        " -F: '{print $2}' | awk '{print $1}'")
+    self.internal_ip = internal_ip.rstrip()
+
+  @vm_util.Retry(poll_interval=10, max_retries=100, log_errors=True)
+  def _ConfigureProxy(self):
+    """
+    In Docker containers environment variables from /etc/environment
+    are not sourced - this results in connection problems when running
+    behind proxy. Prepending proxy environment variables to bashrc
+    solves the problem. Note: APPENDING to bashrc will not work because
+    the script exits when it is NOT executed in interactive shell.
+    """
+    if FLAGS.http_proxy:
+      http_proxy = "sed -i '1i export http_proxy=%s' /etc/bash.bashrc"
+      self.RemoteCommand(http_proxy % FLAGS.http_proxy)
+    if FLAGS.https_proxy:
+      https_proxy = "sed -i '1i export https_proxy=%s' /etc/bash.bashrc"
+      self.RemoteCommand(https_proxy % FLAGS.http_proxy)
+    if FLAGS.ftp_proxy:
+      ftp_proxy = "sed -i '1i export ftp_proxy=%s' /etc/bash.bashrc"
+      self.RemoteCommand(ftp_proxy % FLAGS.ftp_proxy)
+
+  @vm_util.Retry(poll_interval=10, max_retries=100, log_errors=True)
+  def _DeleteApp(self):
+    """
+    Deletes an App.
+    """
+    logging.info('Attempting to delete App: %s' % self.name)
+    output = requests.delete(self.app_url)
+    if output.status_code == requests.codes.NOT_FOUND:
+      logging.info('App %s has been already deleted.')
+      return
+    if output.status_code != requests.codes.OK:
+      raise Exception("Deleting App: %s failed. Reattempting." % self.name)
+
+  def _BuildAppBody(self):
+    """
+    Builds JSON which will be passed as a body of POST request to Marathon
+    API in order to create App.
+    """
+    cat_cmd = ['cat', vm_util.GetPublicKeyPath()]
+    key_file, _ = vm_util.IssueRetryableCommand(cat_cmd)
+    cmd = "/bin/mkdir /root/.ssh; echo '%s' >> /root/.ssh/authorized_keys; " \
+          "/usr/sbin/sshd -D" % key_file
+    body = {
+        'id': self.name,
+        'mem': FLAGS.docker_memory_mb,
+        'cpus': FLAGS.docker_cpus,
```

Comment:
Instead of directly using the docker_memory_mb and docker_cpus flags could you add a MesosVmSpec so that these can be set in configs? Here's an example: https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/providers/gcp/gce_virtual_machine.py#L96